projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -226,6 +297,22 @@ export class TableWidgetRendererComponent
     this.checkboxFilterSubject.next(tableFilter ? [tableFilter] : []);
   }
 
+  public onSelectChange(select: KeyValue<string, unknown>): void {
+    const selectFilterModel = this.model.getSelectFilterOptions().find(option => option.attribute === select.key);
+
+    if (selectFilterModel === undefined) {
+      return; // This shouldn't happen
+    }
+
+    this.selectFilterSubject.pipe(first()).subscribe(filters => {
+      this.selectFilterSubject.next(
+        [...filters.filter(f => f.field !== select.key), selectFilterModel.getTableFilter(select.value)].filter(
```

Comment:
I'm still really unclear what this is doing - removing the selected filter from the list and filtering out unset filters? Why is either necessary? the subscribe looks like it's just to get the current value too, so changing the declared type of `selectFilterSubject` to `BehaviorSubject` (which is already its implementation) will give you direct access to the current value.

Reply:
Yeah, it was just to get the current filters so we can check. I'll use the BehaviorSubject instead and clean this up a bit more.